diffkemp/cli.py

```diff
@@ -12,6 +12,38 @@ def make_argument_parser():
     sub_ap = ap.add_subparsers(dest="command", metavar="command")
     sub_ap.required = True
 
+    # "build" sub-command
+    build_ap = sub_ap.add_parser("build",
+                                 help="build snapshot from Makefile project")
+    build_ap.add_argument("source_dir",
+                          help="project's root directory")
+    build_ap.add_argument("output_dir",
+                          help="output directory of the snapshot")
+    build_ap.add_argument("functions_list", nargs='?',
+                          help="list of functions to compare")
+    build_ap.add_argument("--make", help="make tool used to be used for build",
+                          default="make")
+    build_ap.add_argument("--makefile", help="filename of Makefile to be used\
+                          for build")
```

Comment:
It doesn't have `default=Makefile`, because different versions of Make have their own rules for default makefile when they aren't explicitly given one (e.g. https://www.gnu.org/software/make/manual/html_node/Makefile-Names.html).

As of the default names, they are specific for Make, because generate now only supports Make. I'll change the names to generic ones though to avoid an interface change in the future when support for other tools is added (which it probably will, since Make is slowly becoming kind of obsolete).

Reply:
> It doesn't have `default=Makefile`, because different versions of Make have their own rules for default makefile when they aren't explicitly given one (e.g. https://www.gnu.org/software/make/manual/html_node/Makefile-Names.html).

Still, `Makefile` is "kind of" standard, so I'm slightly in favor of making it default.

> As of the default names, they are specific for Make, because generate now only supports Make. I'll change the names to generic ones though to avoid an interface change in the future when support for other tools is added (which it probably will, since Make is slowly becoming kind of obsolete).

Agreed, let's make them more generic (though I still wouldn't mind having `Makefile` as the default value).

Reply:
> 
> Still, `Makefile` is "kind of" standard, so I'm slightly in favor of making it default.
> 
IMO that would cause unexpected behavior in some cases, e.g. if `GNUmakefile` behaves differently than `Makefile` (either intentionally or unintentionally), `diffkemp build` with default settings would behave differently than `make` (provided `make` is GNU Make).
> Agreed, let's make them more generic (though I still wouldn't mind having `Makefile` as the default value).
With generic UI, it would have to be changed in the future anyway, because e.g. Ninja's defaut build file is `build.ninja` etc.

Reply:
Ok, let's leave it without the default, then.